From 6fce8562105f4975e43bdb495d34427a29ce7bf2 Mon Sep 17 00:00:00 2001
From: vipinc <vipinc@ami.com>
Date: Wed, 14 Sep 2022 21:58:51 +0530
Subject: [PATCH] Enable Threshold Attributes for Core temperature sensors

Enabled Threshold attributes and verified attributes are
available in sysfs path and threshold value is available.

Tested:
Verified the threshold are available for core temp sensors:
root@intel-obmc:~# ipmitool sensor | grep -i core | head -n 10
Core 0 CPU1      | 42.000     | degrees C  | ok    | na        | na        | na        | 79.000    | 87.000    | na
Core 0 CPU2      | 38.000     | degrees C  | ok    | na        | na        | na        | 79.000    | 87.000    | na
Core 1 CPU1      | 39.000     | degrees C  | ok    | na        | na        | na        | 79.000    | 87.000    | na
Core 1 CPU2      | 38.000     | degrees C  | ok    | na        | na        | na        | 79.000    | 87.000    | na
Core 2 CPU1      | 42.000     | degrees C  | ok    | na        | na        | na        | 79.000    | 87.000    | na
Core 2 CPU2      | 37.000     | degrees C  | ok    | na        | na        | na        | 79.000    | 87.000    | na
Core 3 CPU2      | 36.000     | degrees C  | ok    | na        | na        | na        | 79.000    | 87.000    | na
Core 4 CPU1      | 39.000     | degrees C  | ok    | na        | na        | na        | 79.000    | 87.000    | na
Core 4 CPU2      | 37.000     | degrees C  | ok    | na        | na        | na        | 79.000    | 87.000    | na
Core 5 CPU1      | 40.000     | degrees C  | ok    | na        | na        | na        | 79.000    | 87.000    | na

Signed-off-by: vipinc <vipinc@ami.com>
---
 drivers/hwmon/peci-cputemp.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/hwmon/peci-cputemp.c b/drivers/hwmon/peci-cputemp.c
index 5037bce2d85f..26c71b6f753a 100644
--- a/drivers/hwmon/peci-cputemp.c
+++ b/drivers/hwmon/peci-cputemp.c
@@ -72,7 +72,8 @@ static const u32 config_table[] = {
 	HWMON_T_LABEL | HWMON_T_INPUT,
 
 	/* Core temperature - for all core channels */
-	HWMON_T_LABEL | HWMON_T_INPUT,
+	HWMON_T_LABEL | HWMON_T_INPUT | HWMON_T_MAX | HWMON_T_CRIT |
+	HWMON_T_CRIT_HYST,
 
 	/* HBM temperature - for all high bandwidth memory channels */
 	HWMON_T_LABEL | HWMON_T_INPUT | HWMON_T_MAX | HWMON_T_CRIT |
-- 
2.25.1

